controller/src/main/java/io/pravega/controller/store/stream/StreamProperty.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream;
+
+import com.google.common.base.Preconditions;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.io.Serializable;
+
+/**
+ * StreamProperty is an abstraction wrapper over stream properties that can be updated via a workflow.
+ * Namely `configuration` and `truncationRecord`.
```

Comment:
I'm not sure what we are naming here. What's it that `configuration` and `truncationRecord` are referring to?

Reply:
StreamConfiguration and StreamTruncationRecord are two metadata objects that we store for a stream in metadata store.
These objects share a property that they are updatable and we want to control and ensure that while these properties are being updated, a corresponding workflow is performed against the change. And we want to allow only one pending update for each of these properties. 
So we have created a new abstraction called StreamProperty that wraps all such properties for the stream. Presently it adds a property called "updating" flag over the encapsulated property.

Reply:
Better call the classes by their names, the text is confusing.

Reply:
done